components/dashboard/src/user-settings/Preferences.tsx

```diff
@@ -136,46 +149,34 @@ export default function Preferences() {
-                    )}
-
-                    {maySetTimeout.data === true && (
-                        <form onSubmit={saveWorkspaceTimeout}>
-                            <InputField
-                                label="Default Workspace Timeout"
-                                hint={
-                                    <span>
-                                        Use minutes or hours, like <span className="font-semibold">30m</span> or{" "}
-                                        <span className="font-semibold">2h</span>
-                                    </span>
-                                }
-                            >
-                                <div className="flex space-x-2">
-                                    <div className="flex-grow">
-                                        <TextInput
-                                            value={workspaceTimeout}
-                                            placeholder="e.g. 30m"
-                                            onChange={setWorkspaceTimeout}
-                                        />
-                                    </div>
-                                    <Button
-                                        htmlType="submit"
-                                        loading={timeoutUpdating}
-                                        disabled={workspaceTimeout === user?.additionalData?.workspaceTimeout ?? ""}
-                                    >
-                                        Save
-                                    </Button>
+                    <form onSubmit={saveWorkspaceTimeout}>
+                        <InputField
+                            label="Default Workspace Timeout"
+                            hint={
+                                <span>
+                                    Use minutes or hours, like <span className="font-semibold">30m</span> or{" "}
+                                    <span className="font-semibold">2h</span>
+                                </span>
+                            }
+                        >
+                            <div className="flex space-x-2">
+                                <div className="flex-grow">
+                                    <TextInput
+                                        value={workspaceTimeout}
+                                        placeholder="e.g. 30m"
+                                        onChange={setWorkspaceTimeout}
+                                    />
                                 </div>
-                            </InputField>
-                        </form>
-                    )}
+                                <Button
+                                    htmlType="submit"
+                                    loading={timeoutUpdating}
+                                    disabled={workspaceTimeout === user?.additionalData?.workspaceTimeout ?? ""}
+                                >
+                                    Save
```

Comment:
**suggestion**: I'd love to move away from this madness of inline forms we've been using lately like this one and move towards a less fragile UX like the one we recently added in the org-level workspace image, see [relevant discussion](https://gitpod.slack.com/archives/C042X6P86BZ/p1695284226015639) (internal).

In this case, we could easily replace this fragile text input with the following:

| Default | Edit |
|-|-|
| <img width="800" alt="Frame 1846" src="https://github.com/gitpod-io/gitpod/assets/120486/9dfcbfe4-d3ae-49c5-bddd-53d05a37eb97"> | <img width="800" alt="Frame 1847" src="https://github.com/gitpod-io/gitpod/assets/120486/a2e98e5d-0b2d-4384-9ac1-1e875cabafde"> |

If we move this setting to the project or org-level, we could also introduce a read-only version of it, just like we did with the org-level workspace image setting.

| Read-only timeout setting |
|-|
| <img width="800" alt="Frame 1848" src="https://github.com/gitpod-io/gitpod/assets/120486/aaea2501-6ccd-4341-93df-35e1b5f1f218"> |

Reply:
Like that idea!